Dockerfile.vcsim

```diff
@@ -1,4 +1,45 @@
+# Create a builder container
+FROM golang@sha256:f254180c5defa2653955e963fb0626e3d4fbbb162f7cff6490e94607d1d867ff AS build
+WORKDIR /go/src/app
+
+# Create appuser to isolate potential vulnerabilities
+# See https://stackoverflow.com/a/55757473/12429735
+ENV USER=appuser
+ENV UID=10001
+RUN adduser \
+    --disabled-password \
+    --gecos "" \
+    --home "/nonexistent" \
+    --shell "/sbin/nologin" \
+    --no-create-home \
+    --uid "${UID}" \
+    "${USER}"
+
+# Build
+RUN CGO_ENABLED=0 go get -u github.com/vmware/govmomi/vcsim
```

Comment:
Hm, what are you expecting this to do? Go mod versioning would come into your way here...why not build as usual instead of `go get`? (see for example the Dockerfile I gave in the discussion)

Reply:
> what are you expecting this to do?

This downloads and builds the application.

> .why not build as usual instead of go get?

I'm building this inside the `build` container.  Then copying the binary to the `scratch` container.  I don't have `go` on my machine, and I don't expect Dockerhub to or anyone who wants to build this to have it either.

It looks like `Dockerfile.govc` is building on the host, and then copying the binary into the new container.

Am I missing a point you are trying to make?  I'm sorry, I haven't had my coffee yet.  Can you explain further?

Reply:
I see what you mean.  Resolved.